etc/apparmor.d/abstractions/xchat-based

```diff
@@ -53,6 +58,9 @@
 	/usr/lib/*-linux-gnu/** mrix,
 	/usr/lib/xchat/plugins/* mr,
 	/usr/lib/perl*/** mr,
+	/var/lib/snapd/desktop/applications/ r,
+	/usr/lib/firefox-esr/firefox* Ux,
```

Comment:
> U - exec to unconfined, with environment variable scrubbing

Not safe.

> P - exec to profile that matches executable name, with environment variable scrubbing

Sounds better.

Could you change that and test please?

Reply:
> Not safe.

@adrelanos an AppArmor profile for Mozilla Firefox is already available and will ensure a safe execution when enforced. However, it must be customized by the user based on the extensions used.
In this case using `Ux` is not a great weakness, because Mozilla Firefox itself already uses a sandbox.

> Could you change that and test please?

Good idea but I've already tested this combination on multiple computers: without `Ux` Mozilla Firefox won't start ("profile transition not found"). If the Firefox profile is present and enabled, it will be imposed no matter what.
I believe that `Ux` is a reasonable compromise and that, given the particular architecture of Firefox - unlike `ux` - does not compromise security.

Reply:
Debian package apparmor-profiles-extra contains /etc/apparmor.d/usr.bin.pidgin which then uses:

    #include <abstractions/ubuntu-browsers>

(Also usr.bin.evince and usr.bin.thunderbird is using it.)

Looks better / more following convention / supporting more packages. What about that one? Could you try please?

(abstractions/ubuntu-browsers is shipped by Debian apparmor package.)